libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java

```diff
@@ -438,7 +443,14 @@ public void fasterRouteFound(DirectionsRoute directionsRoute) {
     }
   };*/
 
-  private Router.Callback routeEngineCallback = new NavigationViewRouteEngineListener(this);
+  private RoutesObserver routesObserver = new RoutesObserver() {
+    @Override
+    public void onRoutesChanged(@NotNull List<? extends DirectionsRoute> routes) {
+      if (routes.size() > 0) {
+        route.setValue(routes.get(0));
+      }
+    }
+  };
```

Comment:
@LukasPaczos I am still trying to understand why we need the `RoutesObserver`

Reply:
@abhishek1508 if you step through the code, you'll notice that `NavigationMapboxMap#drawRoute` is not called from anywhere. This means that the route will not ultimately be displayed.

That's because the `route` live data object that's stored in the view model has only been updated by now-removed and previously unused `NavigationViewRouteEngineListener`.

Reply:
Using the observer additionally guarantees us that the drawn route will always be the one that was passed to the Nav Native for active guidance, without having to perform any additional testing.

Reply:
👍